### PR TITLE
Bug 1822514: Save lastSyncedTime preference when sync account is succ…

### PR DESCRIPTION
**Pull request by Alexandra-Virvara**
…essful

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1822514

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt
Comment (t-p-white) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt:1]: Should we also update last sync time in onError fun L:463?

File: android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/sync/WorkManagerSyncManager.kt [hunk 1]
```diff
@@ -562,7 +562,13 @@ internal fun getSyncState(context: Context): String? {
...
 }
 
-internal fun setLastSynced(context: Context, ts: Long) {
+/**
+ * Saves the lastSyncedTime to the shared preferences
+ *
+ * @param context the context
+ * @param ts - the last synced time in milliseconds
+ */
+fun setLastSynced(context: Context, ts: Long) {
```
Comment (t-p-white) [inline android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/sync/WorkManagerSyncManager.kt:571]: Thanks for adding some context to this func. Perhaps if `ts` was renamed to something like lastSyncTimeMs we don't need the doc?

Comment (Alexandra-Virvara) [reply android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/sync/WorkManagerSyncManager.kt]: it seems a good idea to rename that parameter to a more suggestive one, but it still needs the documentation as the function became public

File: fenix/app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt [hunk 2]
```diff
@@ -413,7 +414,15 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
                 pref.title = getString(R.string.preferences_sync_now)
                 pref.isEnabled = true
 
-                val time = getLastSynced(requireContext())
+                val lastSyncedTime = getLastSynced(requireContext())
```
Comment (t-p-white) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt:417]: nit: we could move this into a single func as it's doing the same thing in both cases?

**Review by t-p-white — APPROVED** [review]
lgtm :) just the nit suggestion but not a blocker

**Pull request merged**